src/controllers/authController.js

```diff
@@ -0,0 +1,154 @@
+        try {
+            await sendEmail(email, `Your verification OTP is: ${otpCode}`)
+        } catch (emailError) {
+            // If email sending fails, delete the user and OTP from the database
+            await User.deleteOne({ _id: user._id })
+            await Otp.deleteOne({ _id: otp._id })
+
+            console.error('Error sending email:', emailError)
+            return res
+                .status(500)
+                .json({
+                    success: false,
+                    message:
+                        'Failed to send verification email. Please try again later.',
+                })
+        }
+
+        // Respond with success
+        res.json({
+            success: true,
+            message: 'Registration successful. Please verify your email.',
+        })
+    } catch (err) {
+        console.error('Server error:', err)
+        res.status(500).json({
+            success: false,
+            message: 'Server error. Please try again later.',
+        })
+    }
+}
+
+// In controllers/authController.js
+exports.verifyOtp = async (req, res) => {
+    const { email, otp } = req.body
+
+    try {
+        // Find user by email
+        const user = await User.findOne({ email })
+        if (!user)
+            return res
+                .status(404)
+                .json({ success: false, message: 'User not found.' })
+
+        // Find OTP record
+        const otpRecord = await Otp.findOne({
+            userId: user._id,
+            otpCode: otp,
+            isUsed: false,
+        })
+
+        // Validate OTP
+        if (!otpRecord)
+            return res
+                .status(400)
+                .json({ success: false, message: 'Invalid OTP.' })
+        if (otpRecord.expiresAt < new Date())
+            return res
+                .status(400)
+                .json({ success: false, message: 'OTP expired.' })
+
+        // Mark OTP as used
+        otpRecord.isUsed = true
+        await otpRecord.save()
+
+        // Mark the email as verified
+        user.isEmailVerified = true
+        await user.save()
+
+        res.json({ success: true, message: 'Email verified successfully.' })
+    } catch (err) {
+        console.error(err)
+        res.status(500).json({ success: false, message: 'Server error' })
+    }
+}
+
+exports.login = async (req, res) => {
+    const { email, password } = req.body
+    try {
+        const user = await User.findOne({ email })
+        if (!user)
+            return res.status(401).json({
+                success: false,
+                message: 'Invalid credentials',
+                token: null,
+            })
+
+        const isMatch = await bcrypt.compare(password, user.passwordHash)
+        if (!isMatch)
+            return res.status(401).json({
+                success: false,
+                message: 'Invalid credentials',
+                token: null,
+            })
+
+        const token = jwt.sign({ id: user._id }, process.env.JWT_SECRET, {
```

Comment:
JWT logic duplication. Password and token logic lives in `token.util.js`
```suggestion
        const token = generateToken({ userId: user._id })
```